### ISSUE TITLE
Module out-of-date in yotta registry?

### ISSUE THREAD

**Issue by andresag01** [ARMmbed/ble-x-nucleo-idb0xa1#5]
It seems that this module is out-of-date in the yotta registry. As a result it cannot be directly used to build the BLE examples. I get errors like the following:

```
/repos/ble-examples/BLE_EddystoneService/yotta_modules/x-nucleo-idb0xa1/source/BlueNRGGattServer.cpp.o -c /Users/andama01/Documents/repos/ble-examples/BLE_EddystoneService/yotta_modules/x-nucleo-idb0xa1/source/BlueNRGGattServer.cpp
/Users/andama01/Documents/repos/ble-examples/BLE_EddystoneService/yotta_modules/x-nucleo-idb0xa1/source/BlueNRGGattServer.cpp: In member function 'virtual ble_error_t BlueNRGGattServer::addService(GattService&)':
/Users/andama01/Documents/repos/ble-examples/BLE_EddystoneService/yotta_modules/x-nucleo-idb0xa1/source/BlueNRGGattServer.cpp:199:97: error: 'class GattAttribute' has no member named 'getInitialLength'
         if ((p_char->getValueAttribute().getValuePtr() != NULL) && (p_char->getValueAttribute().getInitialLength() > 0)) {
                                                                                                 ^
```

But it seems the issue has been fixed in this repository, [see here](https://github.com/ARMmbed/ble-x-nucleo-idb0xa1/blob/master/source/BlueNRGGattServer.cpp#L199)

**Comment by rgrover** [comment]
Is this an issue of publishing the latest of the module?

**Comment by apalmieriGH** [comment]
The module has been updated in the public repository [ble-x-nucleo-idb0xa1](https://github.com/ARMmbed/ble-x-nucleo-idb0xa1)
Who is in charge of publishing in the yotta registry?

**Comment by andresag01** [comment]
@rgrover: If I clone this repo and manually link the module it builds. The one that yotta automatically pulls in from the registry seems to fail the build stage.

**Comment by rgrover** [comment]
```$ yt owners list x-nucleo-idb0xa1
module "x-nucleo-idb0xa1" owners: wolfgang.betz@st.com, wolfgang.betz@st.com
```

**Comment by apalmieriGH** [comment]
@andresag01 @rgrover @betzw : Commit 1919996 just published to yotta registry

**Comment by rgrover** [comment]
@andresag01 could you please try this?

**Comment by rgrover** [comment]
we now have a successful  build using the latest of this module. :+1:

**Comment by andresag01** [comment]
@apalmieriGH: Thanks for fixing, this works now

**Comment by apalmieriGH** [comment]
why the module automatically pulled from the yt registry is still outdated even after I published it?

**Comment by rgrover** [comment]
did you publish to the public registry? Have you done a `yt update`?

**Comment by apalmieriGH** [comment]
yes, but the module dependency within ble/module.json needs to be updated:

```
"x-nucleo-idb0xa1": "^2.0.1"

```

**Comment by betzw** [comment]
Questo non e' proprio coretto se si considera quello che c'è sul develop branch :-(

On 12/17/2015 04:03 PM, apalmieriGH wrote:

yes, but the module dependency within ble/module.json needs to be updated:

"x-nucleo-idb0xa1": "^2.0.1"

**Comment by rgrover** [comment]
`^2.0.1` means `>= 2.0.1 and < 3.0.0`

**Comment by apalmieriGH** [comment]
ok, so also `2.0.0` should be fine.
Thanks.

**Comment by rgrover** [comment]
no, `2.0.0` is not within `>= 2.0.1 and < 3.0.0`

**Comment by apalmieriGH** [comment]
ok, sorry for misunderstanding the semantic versioning. anyway, the latest published version is:

`^2.0.1`

please, update the ble/module.json accordingly.

**Comment by rgrover** [comment]
module.json in ble says that it needs ^2.0.0 of your library as a target dependency. If you've released v2.0.1, then that should be picked up automatically. Nothing needs to change in ble's module.json until you jump to v3.0 of your module. I don't see what you need here.

**Comment by apalmieriGH** [comment]
ok, `^2.0.0` is fine.
